picamera2/picamera2.py

```diff
@@ -383,7 +382,7 @@ def __exit__(self, exc_type, exc_val, exc_traceback):
 
     def __del__(self):
         """Without this libcamera will complain if we shut down without closing the camera."""
-        self.log.debug(f"Resources now free: {self}")
+        _log.debug(f"Resources now free: {self}")
```

Comment:
Using format strings in logs is generally slightly frowned upon. Modern lint tools will push back on this because it requires a lot of string formatting to happen when the log might never be rendered propagated.